hachoir/listtool.py

```diff
@@ -0,0 +1,101 @@
+#
+# Tool for parsing a file and writing all fields to stdout.
+#
+
+from hachoir.core.cmd_line import getHachoirOptions, configureHachoir
+from hachoir.core.cmd_line import displayVersion
+from hachoir.stream import InputStreamError, FileInputStream
+from hachoir.parser import guessParser, HachoirParserList
+from optparse import OptionGroup, OptionParser
+import sys
+
+
+def printFieldSet(field_set, args, options={}, indent=0):
+    indent_string = "  " * indent
+    for field in field_set:
+        value_display = ""
+        if field.value is not None and options["display_value"]:
+            value_display = f": {field.display}"
+        size_display = ""
+        if options["display_size"]:
+            size_display = f", {field.size}b"
+        print(f"{indent_string}{field.name} <{field.__class__.__name__}{size_display}> ({field.description}){value_display}")
```

Comment:
I would prefer to hide the description by default, and only show it if ``--descr`` (new) option is used.

Reply:
I've added `--description`; the on command line the option names can be shortened as long as they are unique.